### ISSUE TITLE
Reconnect issue

### ISSUE THREAD

**Issue by heleon19** [Supergiovane/node-red-contrib-knx-ultimate#132]
Hello Massimo
How are you? Haven't heard from you for a long time.

We have the problem on two installations that very likely after a network interruption the connection is no longer established correctly.
The nodes are all green, so they do not show any error. However, writing and reading does not work. The watchdog which writes to a group address recognizes the error.
After a deploy, the connection works again.

Attached is the log from this morning, and the configuration of the interface.

We are using the KNX Ultimate 1.2.14, since then there seems to have been no releases that fix my problem.

Thanks for your help

![nodes connected](https://user-images.githubusercontent.com/10326318/128978299-01b07484-c28f-4a34-9f69-3ec6334ee796.png)

![watchdog](https://user-images.githubusercontent.com/10326318/128978317-80038ea6-1aec-46dd-b446-7ef040bf6f22.png)
[watchdog_flows.txt](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6966233/watchdog_flows.txt)
[2021-08-11.log](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6966236/2021-08-11.log)

**Comment by Supergiovane** [comment]
Hi Eleon
i'm fine, thanks.
I'm busy with KNX Secure at the moment.
I saw you're connecting via a KNX Interface in unicast.
Please enable che checkbox "Ignore ACK requests", this one:

<img width="518" alt="Node-RED" src="https://user-images.githubusercontent.com/37666636/128979940-fbd80863-6eb8-4baf-83d3-0368a8e944e0.png">

Then, please, at least on one of the two installations, upgrade to the latest version, because there are everytime little changes/iprovements involving the connection part as well.

**Comment by Zrmite** [comment]
Hi Massimo,
We were able to recreate the problem by removing and plug back in the network cable at the IP interface a few times.
After that, we set the "Ignore ACK requests" and updated to the latest KNX Ultimate Version (1.2.36) and we still get the same result, but now with a new error in the log:
![image](https://user-images.githubusercontent.com/52327137/129010891-68913353-b24d-4853-8486-92e734380aa4.png)
After a deploy it reconnected correctly again.

**Comment by Supergiovane** [comment]
Si Simon,
Every KNX Interface behaves differently, as far i have seen in these years.
As soon as you unplug the cord, some interfaces will release the tunnel after a while, some other interface fail to do so and then you'll run out of free tunneling connections.
For this reason, i ever suggest using a KNX Router.
I tried to reproduce your issue, without success.
For me to check and solve your particular issue, **i need to know the make/model of your KNX interface** on both the installation site.

**Comment by Supergiovane** [comment]
Hi
a Support's person at Weinzierl, was aware of an issue with Weinzierl 731.
See here https://github.com/Supergiovane/node-red-contrib-knx-ultimate/issues/78

Are you able to perform a firmware upgrade on Weinzierl 731?
If not, please leave me today to make other tests..

**Comment by Supergiovane** [comment]
..and another question... do you have more than 1 ethernet/wifi interface on the PC running node-red?

**Comment by Supergiovane** [comment]
Hi Simon
Please unzip this file into your **./node-red/node_modules/node-red-contrib-knx-ultimate/nodes** folder and restart node-red. Then try again unplugging the cable.

[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6968085/knxUltimate-config.js.zip)

**Comment by heleon19** [comment]
11 Aug 14:00:00 - [warn] ------------------------------------------------------
11 Aug 14:00:00 - [warn] [node-red-contrib-knx-ultimate/knxUltimate-config] SyntaxError: Invalid left-hand side in assignment
11 Aug 14:00:00 - [warn] ------------------------------------------------------
11 Aug 14:00:00 - [info] Waiting for missing types to be registered:
11 Aug 14:00:00 - [info]  - knxUltimate-config

With this file I do get an error

**Comment by Supergiovane** [comment]
Hi Heleon,
you need to use the latest version of knx-ultimate, then replace the file.

**Comment by Supergiovane** [comment]
You can revert to a previous version by issuing a command:
npm install node-red-contrib-knx-ultimate@1.2.14

**Comment by Supergiovane** [comment]
Then please use this zip (not the previous) :

[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6968249/knxUltimate-config.js.zip)

**Comment by heleon19** [comment]
This file for the lates version?

**Comment by Supergiovane** [comment]
Yes. Please be aware of the destination folder
./node-red/node_modules/node-red-contrib-knx-ultimate/**nodes**

**Comment by heleon19** [comment]
I was now able to load the file.

We can observe different scenarios when we disconnect the network cable at the interface for different lengths.

1. for longer than 10 seconds, reading nodes turn gray, then green, then blue. The connection is ok. Writing nodes are green.
2. for less than 1 second, reading nodes turn gray, then blue. The writing nodes are gray. However, the connection is ok.
3. for 2-3 seconds, reading nodes turn gray, then green, then gray. Writing nodes are also gray. Connection is not ok, recovers only after a deploy.

**Comment by Supergiovane** [comment]
Please unzip this file into your folder: **.node-red/node_modules/node-red-contrib-knx-ultimate/knxultimate-api2/src**
[FSM.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6968788/FSM.js.zip)

I'm going forward in the blind, because i've tested several times with several disconnection intervals and knx-ultimate always regain the connection and the ability to write to the bus as well.

**Comment by Supergiovane** [comment]
Hi Reto
please unzip this file here **./node-red/node_modules/node-red-contrib-knx-ultimate/nodes** as well:

[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6968861/knxUltimate-config.js.zip)

**Comment by Supergiovane** [comment]
Hi
Please dont' care about the status colors.
For me is important to know: does knx-ultimate reconnects and works ok in each case you mentioned here https://github.com/Supergiovane/node-red-contrib-knx-ultimate/issues/132#issuecomment-896822967 ?

On every reconnection, knx-ultimate should attempt to re-read the values of all nodes set as "read on connection/reconnection". 
Does this happen?

Thanks

**Comment by heleon19** [comment]
Hi
No in some cases after reconnection the values are not re-read. But after deploy it does.
We use a switch to connect PC and KNX Interface. Do you have the same setup? 
Thank you Reto

**Comment by Supergiovane** [comment]
Hi,
yes. I physically disconnect the lan cable coming from the PC.
Ok, let go step by step. I'll take a look at the "reading" after you confirm this: does does knx-ultimate reconnects and works ok in each case you mentioned here #132 (comment) ?

**Comment by heleon19** [comment]
case 1 and 2 --> yes
case 3 --> no
case 4 --> disconnect while reading --> no

**Comment by Supergiovane** [comment]
Hi,
After disconnection/reconnection of the lan cable, does KNX-Ultimate send and receive telegrams from the KNX Bus?
Is that point fixed?

**Comment by heleon19** [comment]
My colleague who is on site has left. Should I go on site tomorrow to test further? Otherwise I will be back on site on Monday.

Could you make an option at the watchdog which does the reconnect. In combination with the TP test that would be pretty safe? If it can be done without, all the better.

**Comment by Supergiovane** [comment]
Hi Reto
try this one into the folder ./node-red/node_modules/node-red-contrib-knx-ultimate/nodes

[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6970000/knxUltimate-config.js.zip)

**Comment by heleon19** [comment]
Hi Massimo
This file with the FSM.js? I can do one short test today. On monday I'm on side, then I can do more tests.
Thank's for your support

**Comment by Supergiovane** [comment]
Hi Heleon
here another try. 
please unzip this file here **./node-red/node_modules/node-red-contrib-knx-ultimate/nodes**
[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6974482/knxUltimate-config.js.zip)

The issue was very naggy, because it involves an old API part, that was bugged.
I tested it at 360° degrees and it should work. Please use the attached zip wit latest knx-ultimate version already installed.

**Comment by heleon19** [comment]
Hi Massimo
I'm not on side, so I'll test it at Monday.

**Comment by Supergiovane** [comment]
Ok. 
The file contains some unwanted "banana" logs, but in the official release, i'll remove that.

**Comment by heleon19** [comment]
Hi Massimo
Thank you for the new version, I've installed v1.2.38.

Unfortunately, not all problems have been solved. Sometimes when I disconnect the network connection, the nodes turn red "Disconnected", when I plug in again at that moment, the nodes turn green, but I can't switch anything, even the inputs are not read.

Further I noticed that the setting "On connect/reconnect read value" is always Unknown. Even after I have set this.
![image](https://user-images.githubusercontent.com/10326318/129529866-96884dd7-f4be-41d6-876c-ad7f4ac05d81.png)

**Comment by Supergiovane** [comment]
Hi Heleon
there should be something wrong if the option "Bei Verbidung/.... der Wert lesen" doesn't remain selected.
Are the node.js and node-red up to date?
Node.js should be at least 12.x or better 14.x, while node-red at least 1.1.x

**Comment by heleon19** [comment]
Node.js v12.16.1 and Node-RED 1.1.3

Just found out, if I select "Lesen den letzten ..." and then go back to "Nein" or "Lesen vom KNX-BUS" it works. So I'think we have to use an upgrade Script for old projects? Or you could parse the value in oneditoredit to 0 or 1?

But the issue with the connection is still our big problem ;-)

**Comment by Supergiovane** [comment]
Hi
If you leave that unselected, it behaves exactly as you set on the old version (There was only yes and no). This behaviour is for backward compatibility and it's the base logic of all changes in my nodes: not to change the setted behaviour.
So you can leave it blank or select option 1 (Lesen vom KNX-BUS).

The issue with connection should be 100% solved, i tested many times with many disconnection attempts and speed (< 500ms, < 1 second, between 1 and 3, < 10 and > than 10 seconds) and it has ever regained the connection (the old version, instead, had the issue you mentioned in the first post).
I suspect an issue with your KNX Interface, as i told you above in this discussion. If you can, you should upgrade it's firmware or try with another KNX Interface.

Are you eventuelly able to make a video (either record the screen or use your cell-phone to make a video of the screen) and send it here?

**Comment by heleon19** [comment]
Just send you an mail with the video. I will not upload it here. Let me know when you get it.

**Comment by Supergiovane** [comment]
Got it.
What happens if you put an inject node here and send the value directly, istead of clicking on the big blue lamp on the right?
<img width="491" alt="Cursor_e_20210816_112827_254009523428740_mp4" src="https://user-images.githubusercontent.com/37666636/129543929-b5de614b-f403-461a-ba94-1951b04b0d2b.png">

**Comment by heleon19** [comment]
It makes no difference, just tested it.

**Comment by Supergiovane** [comment]
I don't know how to help you. 
I cannot reproduce your issue. I checked with 1 Enertex interface, 1 Enertex Router set as interface, and 1 Eibmarkt Interface, all without any issue. Tried the same as you, by detaching, reattaching the interface's RJ connector and all the time, it has regained the connection.
The fact that when you restart node-red, it works again, can be related to some underlying thing happening in the "dgram" component of node.js.
I strongly suggest you to try another KNX IP interface, or even better a KNX router.
As i wrote, your interface had problem in the past.

**Comment by heleon19** [comment]
how is your network setup? As I said, the PC is connected to the interface via a switch. I disconnect the interface, but the PC is still connected to the network. This certainly makes a difference than disconnecting the PC from the network, the detection is different.

I noticed that as long as I wait after an interruption until all values have been read initially, the connection is established correctly. However, if I disconnect again during the initial readout, as shown in the video, the connection is no longer established.

**Comment by Supergiovane** [comment]
Hi Reto
i tested by disconnecting the cable direcly at interrace level, connected by a switch, like your.
Please unzip this and retry:

[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992003/knxUltimate-config.js.zip)

**Comment by Supergiovane** [comment]
If unsuccessful, please unzip this:
[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992029/knxUltimate-config.js.zip)

**Comment by heleon19** [comment]
I tested without the modificated file with an Weinzierl IP Interface 732, still the same as with the 731.
I'll do the test with the modificated file now.

**Comment by Supergiovane** [comment]
Use the latest one, please

**Comment by heleon19** [comment]
Still the same. I disconnect for one two seconds. While the values are then initially read out, I disconnect again, after which the connection never recovered. Can you  put some Banana Log in your test files, so we know the file is correct loaded?

**Comment by heleon19** [comment]
It definitely happens when an interrupt is during the initial readout function.

**Comment by Supergiovane** [comment]
I'm focusing on that.

**Comment by Supergiovane** [comment]
[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992149/knxUltimate-config.js.zip)
Try this.
You should see some BANANA logs

**Comment by heleon19** [comment]
Now the connection is never restored even if I wait for the initial readout.

BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
[info] 8/16/2021, 2:06:44 PM KNXUltimate-config: Loaded saved GA values 37
[info] 8/16/2021, 2:06:44 PM KNXUltimate-config: Do DoInitialReadFromKNXBusOrFile
[info] 8/16/2021, 2:07:03 PM KNXUltimate-config: wrote peristent values to the file /xyz-data/knxultimatestorage/knxpersistvalues/knxpersist6f05cdf2.6fd4bc.json
[warn] 8/16/2021, 2:07:03 PM knxUltimate-config: Disconnected.
[warn] 8/16/2021, 2:07:03 PM knxUltimate-config: knxConnection warning: timed out waiting for CONNECTIONSTATE_RESPONSE
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
[info] 8/16/2021, 2:07:08 PM KNXUltimate-config: Loaded saved GA values 37
[info] 8/16/2021, 2:07:08 PM KNXUltimate-config: Do DoInitialReadFromKNXBusOrFile
[info] 8/16/2021, 2:07:32 PM KNXUltimate-config: wrote peristent values to the file /xyz-data/knxultimatestorage/knxpersistvalues/knxpersist6f05cdf2.6fd4bc.json
[warn] 8/16/2021, 2:07:32 PM knxUltimate-config: Disconnected.
[warn] 8/16/2021, 2:07:32 PM knxUltimate-config: knxConnection warning: timed out waiting for CONNECTIONSTATE_RESPONSE
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
[info] 8/16/2021, 2:07:37 PM KNXUltimate-config: Loaded saved GA values 37
[info] 8/16/2021, 2:07:37 PM KNXUltimate-config: Do DoInitialReadFromKNXBusOrFile
[info] 8/16/2021, 2:07:51 PM KNXUltimate-config: wrote peristent values to the file /xyz-data/knxultimatestorage/knxpersistvalues/knxpersist6f05cdf2.6fd4bc.json
[warn] 8/16/2021, 2:07:51 PM knxUltimate-config: Disconnected.
[warn] 8/16/2021, 2:07:51 PM knxUltimate-config: knxConnection warning: timed out waiting for CONNECTIONSTATE_RESPONSE
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
[info] 8/16/2021, 2:07:57 PM KNXUltimate-config: Loaded saved GA values 37
[info] 8/16/2021, 2:07:57 PM KNXUltimate-config: Do DoInitialReadFromKNXBusOrFile
[info] 8/16/2021, 2:08:25 PM KNXUltimate-config: wrote peristent values to the file /xyz-data/knxultimatestorage/knxpersistvalues/knxpersist6f05cdf2.6fd4bc.json
[warn] 8/16/2021, 2:08:25 PM knxUltimate-config: Disconnected.
[warn] 8/16/2021, 2:08:25 PM knxUltimate-config: knxConnection warning: timed out waiting for CONNECTIONSTATE_RESPONSE
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
BANANA STOP HANDLING TELEGRAMS
[info] 8/16/2021, 2:08:31 PM KNXUltimate-config: Loaded saved GA values 37
[info] 8/16/2021, 2:08:31 PM KNXUltimate-config: Do DoInitialReadFromKNXBusOrFile
16 Aug 14:08:39 - [info] Stopping flows

**Comment by Supergiovane** [comment]
The only thing not covered by a sudden disconnection, was the response sent from some node set to respond to read request.
This is yet covered. Try this:
[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992245/knxUltimate-config.js.zip)

How many nodes are set to "read on start"? 37 as i see in the log?

**Comment by heleon19** [comment]
Still the same. I disconnect for one two seconds. While the values are then initially read out, I disconnect again, after which the connection never recovered.

Yes around 40 values is possible.

**Comment by heleon19** [comment]
Delay between each telegram (in milliseconds) 
50
 and further multiply delay only between -read- telegrams
5x

**Comment by Supergiovane** [comment]
Ok, i'll set same like you and i'll test again. 
Until i cannot reproduce the issue, i'm in the blind.

**Comment by heleon19** [comment]
Yes I understand, maybe we have to spend you some holidays in Switzerland? ;-)

**Comment by Supergiovane** [comment]
Not only mine reconnects after disconnection, but also it performs again the read status.
I'm keeping addin nodes, to see if there is a link between node numbers and issue.

https://user-images.githubusercontent.com/37666636/129565747-2d5bee9e-2301-4052-987d-b786cc01c838.mov

**Comment by heleon19** [comment]
Your nodes change to gray, mine to red

**Comment by Supergiovane** [comment]
Can you send me again your TXT log please?

**Comment by heleon19** [comment]
[2021-08-16.log](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992467/2021-08-16.log)

**Comment by Supergiovane** [comment]
Despite your video was "4K high resolution Dolby video", i wasn't able to read the status of your nodes when they becomes grey.
What do they say?

**Comment by heleon19** [comment]
It never changes to grey! So it changes to red and it says: Disconnected

**Comment by Supergiovane** [comment]
Yes, sorry, my mistake.
Very interesting. After briefly switching to red, the nodes should change to grey and telling you the something line "Will connect again in xx secs." (xx is a random number).
Do you see that?

**Comment by heleon19** [comment]
No it changes from red to green

**Comment by Supergiovane** [comment]
[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992667/knxUltimate-config.js.zip)
Can you try this? I've added some logs to allow me to see what happen.
Then try and i need again your log.

**Comment by heleon19** [comment]
[2021-08-16-2.log](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6992801/2021-08-16-2.log)

**Comment by Supergiovane** [comment]
In the point 2., it lacks the "**Will connect in 6 secs. with ZB xyz**" message, thus the "**ZB xyz connected**" message.
On what are you running node-red? Raspberry PI ?
Do you consider the hardware slow or not?

**Comment by heleon19** [comment]
Yes it runs on a PI, but it all runs fast. It is also rather a small project.

**Comment by Supergiovane** [comment]
Ok, please unzip this and restart
knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6993016/knxUltimate-config.js.zip)
Then add these two nodes:

<img width="439" alt="Node-RED" src="https://user-images.githubusercontent.com/37666636/129578115-fc3f7901-4700-40f6-9f37-40164b61db0f.png">

Then set the STATUS node to report status of only one of your knx-ultimate nodes.
<img width="516" alt="Node-RED" src="https://user-images.githubusercontent.com/37666636/129578253-d3375355-88a8-428b-9040-03196fe684fd.png">

And set the debug node so:
<img width="533" alt="Node-RED" src="https://user-images.githubusercontent.com/37666636/129578383-e74f765b-103a-42f4-9f1f-724378f62f2e.png">

You'll be filled with tons of node statuses in the log. 
Please do a test and post here again the log.
Thanks

**Comment by Supergiovane** [comment]
PS: In the knx-ultimate gateway node, select an ethernet interface, don't leave "AUTO".
<img width="499" alt="Node-RED 2" src="https://user-images.githubusercontent.com/37666636/129578754-e3282c48-e606-469c-b456-7253ccec2693.png">

**Comment by heleon19** [comment]
[2021-08-16-3.log](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6993161/2021-08-16-3.log)

Had to deploy again

**Comment by Supergiovane** [comment]
Ok,
we may be there.
I'll send you two files. 
First, try with this:

[knxUltimate-config.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/6993219/knxUltimate-config.js.zip)

No need for log, only test it and see if it works.

**Comment by heleon19** [comment]
Now it seems to work, what did you change? Until now it has always worked, hope it stays that way.

**Comment by Supergiovane** [comment]
Thanks to your last log, i noticed that, for an unknown reason, seens that the KNX interface fails to raise an error if i send a telegram to it without a proper connection setup.
We'll see it this will continue to work.
I've another file ready for you, with a more specific handling of that "timeout" state, but if it works so, we can postpone this furhter udate.

**Comment by heleon19** [comment]
Thank you, we will test some more times, and report to you. If you made a release of it, I could test it at more systems.

**Comment by Supergiovane** [comment]
Yes, i'll release it tomorrow, because now i must do my life-job. I've about 40 Mails to reply to!
;-)

**Comment by Supergiovane** [comment]
Version 1.2.39

**Comment by heleon19** [comment]
Thank you again! Up to now it seems to work. We will test it on some systems and let you know, if something unexpected happens.
